zcash_client_sqlite/src/lib.rs

```diff
@@ -401,18 +403,24 @@ impl<P: consensus::Parameters> WalletWrite for WalletDb<rusqlite::Connection, P>
         blocks: Vec<ScannedBlock<sapling::Nullifier>>,
     ) -> Result<Vec<Self::NoteRef>, Self::Error> {
         self.transactionally(|wdb| {
-            let start_position = blocks.first().map(|block| {
-                Position::from(
-                    u64::from(block.metadata().sapling_tree_size())
-                        - u64::try_from(block.sapling_commitments().len()).unwrap(),
+            let start_positions = blocks.first().map(|block| {
+                (
+                    block.height(),
+                    Position::from(
+                        u64::from(block.metadata().sapling_tree_size())
+                            - u64::try_from(block.sapling_commitments().len()).unwrap(),
+                    ),
                 )
             });
             let mut wallet_note_ids = vec![];
             let mut sapling_commitments = vec![];
-            let mut end_height = None;
-
+            let mut last_scanned_height = None;
+            let mut note_positions = vec![];
             for block in blocks.into_iter() {
-                if end_height.iter().any(|prev| block.height() != *prev + 1) {
+                if last_scanned_height
+                    .iter()
+                    .any(|prev| block.height() != *prev + 1)
+                {
```

Comment:
```suggestion
                if last_scanned_height.map_or(false, |prev| {
                    block.height() != *prev + 1
                }) {
```
(or however rustfmt formats that). Non-blocking.